packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/plugins/redirects.ts

```diff
@@ -0,0 +1,55 @@
+import { NextRequest, NextResponse } from 'next/server';
+import {
+  RedirectType,
+  GraphQLRedirectsService,
+  REDIRECT_TYPE_PREFIX,
+  REDIRECT_TYPE_DEFAULT,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import config from 'temp/config';
+import { MiddlewarePlugin } from '..';
+
+class RedirectsPlugin implements MiddlewarePlugin {
+  private redirectsService: GraphQLRedirectsService;
+  order = 0;
+
+  constructor() {
+    this.redirectsService = new GraphQLRedirectsService({
+      endpoint: config.graphQLEndpoint,
+      apiKey: config.sitecoreApiKey,
+      siteName: config.jssAppName,
+    });
+  }
+
+  /**
+   * exec async method - to find coincidence in url.pathname and redirects of site
+   * @param req<NextRequest>
+   * @returns Promise<NextResponse>
+   */
+  async exec(req: NextRequest): Promise<NextResponse> {
+    let existsRedirect: RedirectType | undefined;
+    let redirectType: number | undefined;
+    const url = req.nextUrl.clone();
+    // Find the redirect from result of RedirectService
+    const redirects = await this.redirectsService.fetchRedirects();
+
+    if (redirects?.length) {
+      existsRedirect = redirects.find(
+        (redirect: RedirectType) => redirect.pattern === url.pathname
+      );
+    }
+
+    if (existsRedirect) {
+      url.pathname = existsRedirect.target;
+      /** http code of redirect type **/
+      redirectType = existsRedirect.redirectType.includes(REDIRECT_TYPE_PREFIX)
+        ? Number(existsRedirect.redirectType.substring(REDIRECT_TYPE_PREFIX.length))
+        : REDIRECT_TYPE_DEFAULT;
+
+      return NextResponse.redirect(url, redirectType);
```

Comment:
We need to treat redirectType of `SERVER_TRANSFER`  as `NextResponse.rewrite` instead of `redirect`.